src/validators/is-nested.ts

```diff
@@ -1,13 +1,51 @@
 import { composePropertyDecorators, IsNestedOptions } from '@hippo-oss/dto-decorators';
-import { ValidateNested } from 'class-validator';
+import {
+    ValidateNested,
+    ValidatorConstraintInterface,
+    Validate,
+    ValidationArguments,
+    ValidatorConstraint,
+} from 'class-validator';
 
 import { createBasePropertyDecorators } from './base';
 
+@ValidatorConstraint({ name: 'IsOneOf', async: false })
+export class IsOneOf implements ValidatorConstraintInterface {
+    validate(value: unknown, args: ValidationArguments) {
+        // allow to be handled by normal ValidateNested
+        if (typeof value !== 'object' || value === null) {
+            return true;
+        }
+
+        const definedProperties = Object.values(value)
+            .filter((property) => property !== undefined && property !== null);
```

Comment:
This could be done in one loop but thought readability would be better than slightly faster here?.